doc/design/draft/subscription.md

```diff
@@ -0,0 +1,180 @@
+
+| Publisher | Timepoint | Sample |
+|:---------:|:---------:|:------:|
+| P2        | 42        | J      |
+| P1        | 41        | E      |
+| P2        | 40        | I      |
+| P1        | 31        | D      |
+| P2        | 24        | H      |
+| P2        | 22        | G      |
+
+#### Subscription with _HistorySize_ 20
+
+The subscriber cache contains the ten samples ordered after _DeliveryTime_.
+It contains only ten since there are no more samples available on publisher side.
+
+| Publisher | Timepoint | Sample |
+|:---------:|:---------:|:------:|
+| P2        | 42        | J      |
+| P1        | 41        | E      |
+| P2        | 40        | I      |
+| P1        | 31        | D      |
+| P2        | 24        | H      |
+| P2        | 22        | G      |
+| P1        | 21        | C      |
+| P2        | 20        | F      |
+| P1        | 11        | B      |
+| P1        | 1         | A      |
+
+### Subscribing with concurrent send
+
+Let's assume publisher P1 would like to deliver the sample K at timepoint
+52 while the subscriber requested subsription. Furthermore, the subscriber
+requests always a _HistorySize_ of 2.
+
+All the listed cases should behave identical if there is more then one
+concurrent publish in progress.
+
+#### Publisher before _DeliveryTime_
+
+The subscriber cache contains two samples.
+
+| Publisher | Timepoint | Sample |
+|:---------:|:---------:|:------:|
+| P2        | 42        | J      |
+| P1        | 41        | E      |
+
+The sample K will be delivered after successful subscription.
+
+#### Publisher after _DeliveryTime_
+
+The subscriber cache contains two samples.
+
+| Publisher | Timepoint | Sample |
+|:---------:|:---------:|:------:|
+| P1        | 51        | K      |
+| P2        | 42        | J      |
+
+The sample K will not be delivered by the publisher since the _DeliveryTime_
+is defined as exactly the timepoint after the sample is placed into the ChunkHistory
+and the subscriber cache is based on that ChunkHistory.
+
+### Subscribing After Publisher Stops Offering
+
+In this case we assume that publisher P1 has stopped offering the service
+before the subscription was requested.
+
+#### Subscription with _HistorySize_ 4
+
+The subscriber cache contains the four samples ordered after _DeliveryTime_.
+
+| Publisher | Timepoint | Sample |
+|:---------:|:---------:|:------:|
+| P2        | 42        | J      |
+| P1        | 41        | E      |
+| P2        | 40        | I      |
+| P1        | 31        | D      |
+
+
+
+## Design
+
+### Idea 1: One Ring Buffer Which Contains The Samples
+
+- All publishers are sharing one random access lock-free ring-buffer.
+- The ring-buffer is filled with shared-chunks which can be copied in a threadsafe
+    and lock-free manner.
+- A publisher delivers a sample by writing that sample into the shared ring-buffer.
+- Every subscriber owns an index to their own read position of the ring-buffer.
```

Comment:
Probably not necessary if we extend the idea I have in mind for  #857, but we will see this in a prototype implementation. Any subscriber can read randomly from the cache. It is the concurrent writing (concurrency between writes) which complicates things, I need to explore that.